### ISSUE TITLE
Crash when encountering Google Pay error

### ISSUE THREAD

**Issue by mrsasha** [braintree/braintree_android#186]
### General information

* SDK/Library version: `2.8.1` (by way of Android Drop-in)
* Environment: `Sandbox`
* Android Version and Device: Samsung S8, Android 7.1.2

### Issue description

While updating your Drop-In to 3.2.0 (that uses 2.8.1), I tested Google Pay payments with an unsupported country (German credit card) and an error from Google was shown (`An unexpected error has occurred. Please try again later. [OR-CCSEH-18]`). Upon closing the error dialog, the app crashed. Here's the stacktrace:

```
java.lang.RuntimeException: Parcelable encountered IOException writing serializable object (name = com.braintreepayments.api.exceptions.GooglePaymentException)
      at android.os.Parcel.writeSerializable(Parcel.java:1536)
      at android.os.Parcel.writeValue(Parcel.java:1484)
      at android.os.Parcel.writeArrayMapInternal(Parcel.java:733)
      at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1408)
      at android.os.Bundle.writeToParcel(Bundle.java:1133)
      at android.os.Parcel.writeBundle(Parcel.java:773)
      at android.content.Intent.writeToParcel(Intent.java:9253)
      at android.app.ActivityManagerProxy.finishActivity(ActivityManagerNative.java:3905)
      at android.app.Activity.finish(Activity.java:5369)
      at android.app.Activity.finish(Activity.java:5387)
      at com.braintreepayments.api.dropin.DropInActivity.finish(DropInActivity.java:372)
      at com.braintreepayments.api.dropin.BaseActivity.finish(BaseActivity.java:65)
      at com.braintreepayments.api.dropin.DropInActivity$4.onAnimationFinished(DropInActivity.java:194)
      at com.braintreepayments.api.dropin.DropInActivity$9.onAnimationEnd(DropInActivity.java:358)
      at android.view.animation.Animation$3.run(Animation.java:381)
      at android.os.Handler.handleCallback(Handler.java:751)
      at android.os.Handler.dispatchMessage(Handler.java:95)
      at android.os.Looper.loop(Looper.java:154)
      at android.app.ActivityThread.main(ActivityThread.java:6776)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
   Caused by: java.io.NotSerializableException: com.google.android.gms.common.api.Status
      at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1224)
      at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1584)
      at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1549)
      at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1472)
      at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1218)
      at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:346)
      at android.os.Parcel.writeSerializable(Parcel.java:1531)
      at android.os.Parcel.writeValue(Parcel.java:1484) 
      at android.os.Parcel.writeArrayMapInternal(Parcel.java:733) 
      at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1408) 
      at android.os.Bundle.writeToParcel(Bundle.java:1133) 
      at android.os.Parcel.writeBundle(Parcel.java:773) 
      at android.content.Intent.writeToParcel(Intent.java:9253) 
      at android.app.ActivityManagerProxy.finishActivity(ActivityManagerNative.java:3905) 
      at android.app.Activity.finish(Activity.java:5369) 
      at android.app.Activity.finish(Activity.java:5387) 
      at com.braintreepayments.api.dropin.DropInActivity.finish(DropInActivity.java:372) 
      at com.braintreepayments.api.dropin.BaseActivity.finish(BaseActivity.java:65) 
      at com.braintreepayments.api.dropin.DropInActivity$4.onAnimationFinished(DropInActivity.java:194) 
      at com.braintreepayments.api.dropin.DropInActivity$9.onAnimationEnd(DropInActivity.java:358) 
      at android.view.animation.Animation$3.run(Animation.java:381) 
      at android.os.Handler.handleCallback(Handler.java:751) 
      at android.os.Handler.dispatchMessage(Handler.java:95) 
      at android.os.Looper.loop(Looper.java:154) 
      at android.app.ActivityThread.main(ActivityThread.java:6776) 
      at java.lang.reflect.Method.invoke(Native Method) 
      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520) 
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410) 
```

**Comment by bluk** [comment]
We will fix the error handling in a future update. Thanks for the report.

**Comment by bluk** [comment]
This should be fixed in [Android Drop-in 3.2.1](https://github.com/braintree/braintree-android-drop-in/releases/tag/3.2.1). Thanks.
